internal/security/secret.go

```diff
@@ -90,20 +90,29 @@ func (s *SecretProvider) initializeSecretClient(
 	}
 
 	if s.isSecurityEnabled() {
-		for i := 0; i < secretConfig.AdditionalRetryAttempts; i++ {
-			secretClient, err = client.NewVault(ctx, secretConfig, s.loggingClient).Get(s.configuration.SecretStoreExclusive)
+		secretClient, err = client.NewVault(ctx, secretConfig, s.loggingClient).Get(secretStoreInfo)
+
+		if err == nil || secretConfig.AdditionalRetryAttempts <= 0 {
```

Comment:
Shouldn't this be `if err != ni || ...`

Reply:
no, I am doing retry logic after this. so if it is successfully created without error (ie `err == nil`) then we can return right away